pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -4554,10 +4555,28 @@ protected void internalGetLastMessageId(AsyncResponse asyncResponse, boolean aut
 
     protected void handleTopicPolicyException(String methodName, Throwable thr, AsyncResponse asyncResponse) {
         Throwable cause = thr.getCause();
-        if (!(cause instanceof WebApplicationException)
-                || !(((WebApplicationException) cause).getResponse().getStatus() == 307)) {
-            log.error("[{}] Failed to perform {} on topic {}",
-                    clientAppId(), methodName, topicName, cause);
+
+        Level logLevel = null;
+        if (cause instanceof WebApplicationException) {
+            int statusCode = ((WebApplicationException) cause).getResponse().getStatus();
+            if (statusCode == Status.NOT_FOUND.getStatusCode()) {
+                logLevel = Level.WARN;
+            } else if (statusCode != Status.TEMPORARY_REDIRECT.getStatusCode()) {
+                logLevel = Level.ERROR;
+            }
+        } else {
+            logLevel = Level.ERROR;
+        }
+
+        if (logLevel != null) {
+            switch (logLevel) {
+                case WARN:
+                    log.warn("[{}] Failed to perform {} on topic {}", clientAppId(), methodName, topicName, cause);
```

Comment:
It seems like we could avoid logging the stack trace. What do you think @nodece?

Reply:
Yes, we can avoid this.